apollo-router/src/plugins/coprocessor/mod.rs

```diff
@@ -77,19 +78,26 @@ impl Plugin for CoprocessorPlugin<HTTPClientService> {
             .with_native_roots()
             .with_no_client_auth();
 
-        let connector = hyper_rustls::HttpsConnectorBuilder::new()
+        let builder = hyper_rustls::HttpsConnectorBuilder::new()
             .with_tls_config(tls_config)
             .https_or_http()
-            .enable_http1()
-            .enable_http2()
-            .wrap_connector(http_connector);
+            .enable_http1();
+
+        let connector = if init.config.traffic_shaping.is_none() || init.config.traffic_shaping.as_ref().unwrap().experimental_http2 != Some(Http2Config::Disable) {
+            builder.enable_http2().wrap_connector(http_connector)
+        } else {
+            builder.wrap_connector(http_connector)
+        };
 
         let http_client = ServiceBuilder::new()
             .layer(TimeoutLayer::new(init.config.timeout))
             .service(
                 hyper::Client::builder()
+                    .http2_only(
+                        init.config.traffic_shaping.is_some() && init.config.traffic_shaping.as_ref().unwrap().experimental_http2 == Some(Http2Config::Http2Only)
```

Comment:
Maintain default behavior.  Only set http2_only if traffic_shaping is provided and the experimental_http2 is set to http2only